pkg/sql/sem/builtins/builtins.go

```diff
@@ -3546,6 +3546,37 @@ value if you rely on the HLC for accuracy.`,
 		},
 	),
 
+	"jsonb_array_to_string_array": makeBuiltin(arrayProps(),
```

Comment:
out of curiosity, did you consider implementing this using a SQL body instead? for example:
```
root@localhost:26257/system/defaultdb> select array_agg(x) from (select jsonb_array_elements_text('[1,2,3]') as x);
  array_agg
-------------
  {1,2,3}
```

it might be faster in some cases, but more importantly, it might be easier to maintain.

Reply:
That is how I had initially in this PR, but then Yahor [noticed](https://github.com/cockroachdb/cockroach/pull/112350#pullrequestreview-1680613676) it was causing an apply join that could be concerning, so this is why the suggestion to create the new builtin (which was reviewed on https://github.com/cockroachdb/cockroach/pull/112609 and rebased here so it could make sure it was working as expected)

Reply:
oh got it, ignore me then! at least i learned something :)

Reply:
No worries! I also learned 😄